ably/realtime_channel_test.go

```diff
@@ -151,3 +151,43 @@ func TestRealtimeChannel_AttachWhileDisconnected(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// When publishing a message to a channel, data can be either a single string or
+// a struct of type Message. This example shows both of these ways to publish a message.
+func ExampleRealtimeChannel_Publish() {
+
+	// Create a new realtime client.
+	client, err := ably.NewRealtime(
+		ably.WithKey("ABLY_PRIVATE_KEY"),
+		ably.WithClientID("Client A"),
+	)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
This should return after printing the error

Reply:
For consistency, I've updated all examples to return if there is an error.